### PR TITLE
Feature/authentication backend

### PR DESCRIPTION
**Pull request by ahmed-ben10**

### CODE REVIEW

File: packages/client/src/app/stores/auth.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      await axios
+        .post('/user/logout')
+        .then(() => {
+          router.push({ name: 'home' })
+        })
+        .catch((error) => {
+          console.error(error)
+        })
+    },
+    async getLoggedInUser() {
```
Comment (NoNamer777) [inline packages/client/src/app/stores/auth.js:48]: I am not sure what the purpose of this function will be?

**Review by NoNamer777 — CHANGES_REQUESTED** [review]
Looks very nice, haven't tested it fully yet, will do that later. Left some comments. Well done so far! 👌🏼 👌🏼

File: packages/server/src/app/middleware/cors-middleware.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+let cors = require('cors')
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/middleware/cors-middleware.js:1]: You're right

File: packages/client/src/app/views/RegisterView.vue [hunk 3]
```diff
@@ -1,57 +1,56 @@
...
-import InputLabel from '../components/form/InputLabel.vue'
+import { useAuthStore } from '../stores'
+const { register } = useAuthStore()
 
 const user = ref({
   email: '',
   username: '',
   password: '',
-  passwordRepeat: ''
+  password_confirm: ''
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/client/src/app/views/RegisterView.vue:12]: You're right. Will change it to camelCase

File: packages/client/src/app/views/HomeView.vue [hunk 4]
```diff
@@ -7,7 +13,13 @@
       suscipit. Aliquam architecto corporis cum deleniti doloribus fugiat perspiciatis, qui quis
       ratione repellendus! Cum eaque est provident rerum tempore! Beatae maiores pariatur placeat?
     </p>
-    <router-link class="btn btn-primary" :to="{ name: 'register' }"> Registreer </router-link>
+    <router-link
+      v-if="isAuthenticated"
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/client/src/app/views/HomeView.vue:17]: Yes, should be the other way around. Will fix it

File: packages/server/src/app/models/user/user.entity.js [hunk 5]
```diff
@@ -56,4 +61,15 @@ UserEntity.init(UserModelDefinition, {
   updatedAt: false
 })
 
+UserEntity.prototype.validPassword = function (password) {
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/models/user/user.entity.js:64]: No, i was experimenting a bit. Will remove the prototype part

[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/models/user/user.entity.js:64]: Ohh discovered it works only when it's there 🙃

File: packages/server/src/app/models/user/user.service.js [hunk 6]
```diff
@@ -16,8 +16,6 @@ class UserService {
   /** @type {UserService} */
   static #instance
 
-  // TODO: Do not return user's email and password
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/models/user/user.service.js:19]: The email part not yet. Will add it

File: packages/server/src/app/validatior/index.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+const { validationResult } = require('express-validator')
+
+/**
+ * Checks if the provided string object only contains numbers
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/validatior/index.js:4]: Whoops, will update it

File: packages/server/src/app/services/error-handler.service.js [hunk 8]
```diff
@@ -3,7 +3,8 @@ const InternalServerErrorException = require('../models/errors/internal-server.e
 class ErrorHandlerService {
   #defaultError = new InternalServerErrorException()
 
-  handleError(error, _request, response) {
+  /* eslint-disable  no-unused-vars */
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/services/error-handler.service.js:6]: You're right. Fixed it

File: packages/server/src/app/models/user/user.controller.js [hunk 9]
```diff
@@ -83,6 +85,24 @@ class UserController {
     }
     await UserService.instance().deleteById(parseInt(userIdParam))
   }
+
+  /**
+   * @param data
+   * @return {Promise<UserEntity>}
+   */
+  async auth(data) {
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/models/user/user.controller.js:93]: Moved it to a seperate route & controller

File: packages/client/src/app/stores/auth.js (hunk 1 above)
Comment (ahmed-ben10) [reply packages/client/src/app/stores/auth.js:48]: this get's the authenticated user and with this we can check if someone is authenticated

Comment (ahmed-ben10) [reply packages/client/src/app/stores/auth.js:48]: I renamed it to make it clearer

File: packages/server/src/app/models/user/user.router.js [hunk 10]
```diff
@@ -1,25 +1,68 @@
...
+const { cookieJwtAuth } = require('../../middleware/cookie-jwt-auth')
 
-router.get('/', async (_, response) => {
+router.get('/', cookieJwtAuth, async (_, response) => {
   response.send(await UserController.instance().getAll())
 })
 
-router.get('/:userId', async (request, response) => {
+router.get('/logged_in', cookieJwtAuth, async (request, response, next) => {
+  try {
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/models/user/user.router.js:14]: Yes, but it does it with the authenticated user.

File: packages/server/src/app/models/user/user.router.js [hunk 11]
```diff
@@ -1,25 +1,68 @@
...
+    next(error)
+  }
 })
 
-router.delete('/:userId', async (request, response) => {
+router.delete('/:userId', cookieJwtAuth, async (request, response) => {
   response.send(await UserController.instance().deleteById(request.params.userId))
 })
 
+router.post('/auth', authValidation, async (request, response, next) => {
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/server/src/app/models/user/user.router.js:45]: done!

File: packages/client/src/app/axios/index.js [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        title: response.data.message,
+        type: 'success'
+      })
+    }
+
+    return response
+  },
+  (error) => {
+    const response = error.response
+    if (response.status >= 300 && response.data.errors) {
```
[the remark this answers is not in the document]
Comment (ahmed-ben10) [reply packages/client/src/app/axios/index.js:22]: True, will update it to 400 >

**Review by NoNamer777 — CHANGES_REQUESTED** [review]
Looks good, got some comments before we can merge.

- ~I think you have duplicated the Login and Register view per accident when merging the main branch to fix the merge conflicts?~ (Looks like you found those already 😉)
- Can you check if there are linting/format errors and fix those as well?

**Comment by ahmed-ben10** [comment]
Fixed linting issue`

**Review by NoNamer777 — DISMISSED** [review]
Lookin' Goooddd 👌🏼 🎉

**Review by NoNamer777 — APPROVED** [review]

**Pull request merged**